src/bin/compute-gop.cc

```diff
@@ -174,33 +178,50 @@ int main(int argc, char *argv[]) {
       }
     }
 
-    RandomAccessInt32VectorReader alignment_reader(alignments_rspecifier);
+    RandomAccessInt32VectorReader phoneme_alignments_reader(phoneme_alignments_rspecifier);
+    RandomAccessInt32VectorReader transition_alignments_reader(transition_alignments_rspecifier);
     SequentialBaseFloatMatrixReader prob_reader(prob_rspecifier);
     PosteriorWriter gop_writer(gop_wspecifier);
     BaseFloatVectorWriter feat_writer(feat_wspecifier);
 
     int32 num_done = 0;
     for (; !prob_reader.Done(); prob_reader.Next()) {
       std::string key = prob_reader.Key();
-      if (!alignment_reader.HasKey(key)) {
-        KALDI_WARN << "No alignment for utterance " << key;
+      if (!phoneme_alignments_reader.HasKey(key)) {
+        KALDI_WARN << "No phoneme alignment for utterance " << key;
         continue;
       }
-      auto alignment = alignment_reader.Value(key);
+      if (!transition_alignments_reader.HasKey(key)) {
+        KALDI_WARN << "No transition alignment for utterance " << key;
+        continue;
+      }
+      auto phoneme_alignment = phoneme_alignments_reader.Value(key);
+      auto transition_alignment = transition_alignments_reader.Value(key);
       Matrix<BaseFloat> &probs = prob_reader.Value();
       if (log_applied) probs.ApplyExp();
 
+      std::vector<std::vector<int32> > split;
+      SplitToPhones(trans_model, transition_alignment, &split);
+
+      std::vector<int32> phone_boundary;
+      for (int32 i = 0; i < split.size(); i++) {
+        for (int32 j = 0; j < split[i].size(); j++) {
+          phone_boundary.push_back(static_cast<int32>(i));
```

Comment:
`i` is already of type `int32` and there is no need to cast it to `int32`.

Reply:
OK, I got it. I will remove 'static_cast<int32>'